### DISCUSSION TITLE
Providing the union-default-graph query parameter

### DISCUSSION THREAD

**Discussion by JoelBender** [oxigraph/oxigraph#632] [Q&A] [answered]
According to the [0.2.4](https://github.com/oxigraph/oxigraph/blob/main/CHANGELOG.md#024---2021-04-28) change log there is a way to provide the `union-default-graph` query parameter, is there a SPARQL sample that shows this being provided?  I'm using RDFLib with the SPARQLUpdateStore API and I would like to query it like an RDFLib ConjunctiveGraph, a union of all of the named graphs.

**Comment by Tpt** [comment, marked as answer]
I don't think there is an example with rdflib out there yet. Here it is:

```python
from rdflib.plugins.stores.sparqlstore import SPARQLUpdateStore

store = SPARQLUpdateStore(params={'union-default-graph': '', 'using-union-graph': ''}) # using-union-graph is only if you want the union graph for the WHERE part of INSERT/DELETE WHERE updates
store.open(("http://localhost:7878/query", "http://localhost:7878/update"))
store.update("PREFIX ex: <http://example.com/> INSERT DATA { GRAPH ex:g { ex:s ex:p ex:o }}")
print(list(store.query("SELECT * WHERE { ?s ?p ?o }")))
```

**Reply by JoelBender** [reply]
Excellent, thank you.
